src/lib/proxy/proxy-repository.ts

```diff
@@ -85,11 +88,19 @@ export class ProxyRepository
 
     stop(): void {
         this.stores.eventStore.off(ANY_EVENT, this.onAnyEvent);
-        clearTimeout(this.timer);
+        this.running = false;
```

Comment:
Nice catch; this would be correct if we actually used a setTimeout. In this example we are setting up new timeouts. 

Q: Should this have been a scheduled job?

Reply:
I think the setTimeout is in place here so that we can introduce jitter on the db calls. Otherwise we risk hammering the db at at once when there's multiple active frontend tokens

I'm not quite sure a scheduled job fits here. That seems to be designed to tackle setting up schedules for service level tasks that are consistent and live for the application lifetime. This is a set of schedules on the repository layer with jitter that are added and removed on an adhoc basis depending on what's happening with the frontend tokens that you have active.